sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/ConfigurationServiceSerializer.cs

```diff
@@ -169,18 +169,18 @@ private static void ReadPropertyValue(ConfigurationSetting setting, JsonProperty
             }
         }
 
-        public static async Task<ConfigurationSetting> DeserializeSettingAsync(Stream content, CancellationToken cancellation)
+        public static async Task<ConfigurationSetting> DeserializeSettingAsync(BinaryData content, CancellationToken cancellation)
         {
-            using (JsonDocument json = await JsonDocument.ParseAsync(content, default, cancellation).ConfigureAwait(false))
+            using (JsonDocument json = await JsonDocument.ParseAsync(content.ToStream(), default, cancellation).ConfigureAwait(false))
             {
                 JsonElement root = json.RootElement;
                 return ReadSetting(root);
             }
         }
 
-        public static ConfigurationSetting DeserializeSetting(Stream content)
+        public static ConfigurationSetting DeserializeSetting(BinaryData content)
         {
-            using JsonDocument json = JsonDocument.Parse(content, default);
+            using JsonDocument json = JsonDocument.Parse(content.ToMemory(), default);
             JsonElement root = json.RootElement;
             return ReadSetting(root);
```

Comment:
Why do we pass content.ToMemory() in the synchronous method and content.ToStream() in the asynchronous method when using JsonDocument.Parse? Can we use the same pattern for both methods?

Reply:
I wish that we could use `ToMemory` for both.  The async version requires a stream, which means that we have to wrap the internal `_bytes` from `BinaryData` into a memory stream.  Because the synchronous version doesn't have that limitation, we can pass the bytes directly and avoid the overhead of an allocation and stream wrapper.